### ISSUE TITLE
Confusing and missing Deprecation-/FutureWarnings for ImageContainer and pyresample.bilinear

### ISSUE THREAD

**Issue by ameraner** [pytroll/pyresample#323]
#### Code Sample, a minimal, complete, and verifiable piece of code

```python
import numpy as np
from pyresample import image, geometry
area_def = geometry.AreaDefinition('areaD', 'Europe (3km, HRV, VTC)', 'areaD',
                               {'a': '6378144.0', 'b': '6356759.0',
                                'lat_0': '50.00', 'lat_ts': '50.00',
                                'lon_0': '8.00', 'proj': 'stere'},
                               800, 800,
                               [-1370912.72, -909968.64,
                                1029087.28, 1490031.36])
msg_area = geometry.AreaDefinition('msg_full', 'Full globe MSG image 0 degrees',
                               'msg_full',
                               {'a': '6378169.0', 'b': '6356584.0',
                                'h': '35785831.0', 'lon_0': '0',
                                'proj': 'geos'},
                               3712, 3712,
                               [-5568742.4, -5568742.4,
                                5568742.4, 5568742.4])
data = np.ones((3712, 3712))
msg_con_quick = image.ImageContainerNearest(data, msg_area, radius_of_influence=1000)

```
#### Problem description
A few issues are present on the Deprecation warnings: 
- Initialising `ImageContainerNearest`, `ImageContainerQuick`, etc. initializes `ImageContainer`, which triggers a FutureWarning saying that `NumpyResamplerBilinear` should be used instead. This is a bit confusing for a user that doesn't want to use bilinear resampling, but another of the available algorithms. The FutureWarning should rather be something like `The pyresample.image module is deprecated. Please use pyresample.kd_tree or pyresample.bilinear instead.`, as given in the documentation note: https://pyresample.readthedocs.io/en/latest/grid.html#pyresample-image
- The note of the previous link is present only in the documentation page for gridded data, but is missing for the swath data here 
https://pyresample.readthedocs.io/en/latest/swath.html#pyresample-image.
- In several places ([here](https://github.com/pytroll/pyresample/blob/master/pyresample/bilinear/_numpy_resampler.py#L92) [here](https://github.com/pytroll/pyresample/blob/master/pyresample/bilinear/_numpy_resampler.py#L137) [here](https://github.com/pytroll/pyresample/blob/master/pyresample/bilinear/_numpy_resampler.py#L217)) FutureWarnings indicate the usage of  `NumpyResamplerBilinear`. However, that class is deprecated and triggers a warning indicating to use `NumpyBilinearResampler`. It would be better if the updated class would be indicated in all the warnings. 
- 
#### Expected Output
Updated and correct Deprecation/FutureWarnings

#### Actual Result, Traceback if applicable
```
/tcenas/home/andream/code/fire_val_packages/pyresample/pyresample/image.py:62: FutureWarning: Usage of ImageContainer is deprecated, please use NumpyResamplerBilinear class instead
  FutureWarning)
```

#### Versions of Python, package at hand and relevant dependencies
Pyresample 1.17.0

**Comment by djhoese** [comment]
I think the idea by @pnuu (who added most of these warnings I think) was to deprecate the ImageX interfaces altogether. It seems that they need to be made specific for bilinear resampling for now until we have fully moved to better "Resampler" interfaces.

**Comment by pnuu** [comment]
Yep, the idea was to deprecate all the `ImageContainer` interfaces and point the users to the XArray resampler classes.
